cmd/shortener/main.go

```diff
@@ -40,13 +40,17 @@ func main() {
 	h := handlers.MakeShortanerHandler(c, storage)
 
 	r := chi.NewRouter()
+	r.Use(chiMiddleware.Timeout(60 * time.Second))
 	r.Use(middleware.WithLogRequest)
 	r.Use(middleware.GzipMiddleware)
-	r.Use(chiMiddleware.Timeout(60 * time.Second))
 
-	r.Post("/", h.CreateURL)
-	r.Post("/api/shorten", h.CreateURLByJSON)
-	r.Post("/api/shorten/batch", h.CreateURLByJSONBatch)
+	r.With(middleware.AuthSet).Post("/", h.CreateURL)
+	r.With(middleware.AuthSet).Post("/api/shorten", h.CreateURLByJSON)
+	r.With(middleware.AuthSet).Post("/api/shorten/batch", h.CreateURLByJSONBatch)
+
+	r.With(middleware.AuthCheck).Get("/api/user/urls", h.UserURLs)
+	r.With(middleware.AuthCheck).Delete("/api/user/urls", h.DeleteURLs)
```

Comment:
Можно сгруппировать маршруты:
```suggestion
r.Route("/api", func(r chi.Router) {
    r.Route("/shorten", func(r chi.Router) {
        r.With(middleware.AuthSet).Post("/", h.CreateURLByJSON)
        r.Post("/batch", h.CreateURLByJSONBatch)
    })
    r.Route("/user", func(r chi.Router) {
        r.With(middleware.AuthCheck).Get("/urls", h.UserURLs)
        r.Delete("/urls", h.DeleteURLs)
    })
})

```

Reply:
Сгруппировал с учетом мидлварей